app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt

```diff
@@ -62,13 +62,26 @@ class DrawerActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedLi
         setupDrawerAndToolbar()
 
         prepareForScan()
+
+        val bundle = intent.extras
+        if (bundle?.getString("clipart").equals("clipart")) {
+            val clipart = SavedClipartFragment()
+            supportFragmentManager.beginTransaction()
+                    .replace(R.id.frag_container, clipart)
+                    .commit()
```

Comment:
```suggestion
                    .commit()
            nav_view.setCheckedItem(R.id.saved_cliparts)
```

you also should check the navigation drawer when switching fragments as it needs to reflect as soon as you switch to another fragment.